auslib/blobs/base.py

```diff
@@ -55,6 +55,51 @@ def createBlob(data):
     return blob_map[schema_version](**data)
 
 
+def merge_lists(*lists):
+    result = []
+    for l in lists:
+        for i in l:
+            if i not in result or not isinstance(i, type(result[result.index(i)])):
+                result.append(i)
```

Comment:
This is where the lists of partials and completes get built up, I think, so what does the isinstance check do ? It looks kinda like a str vs unicode special case for hypothesis.

Reply:
Yep, pretty much. merge_lists and merge_dicts are written more generically than is purely necessary. Do you think it would be preferable to target them more, and adjust the tests to remove more cases that we know we won't encounter?

Reply:
I'd be happy if there was a comment to provide some explanation.

Reply:
I added docstrings to merge_lists and merge_dicts in https://github.com/mozilla/balrog/commit/70e07aae4d39854c0d5b1323cf4b5e75c5f0c680.